### DISCUSSION TITLE
Reliable way to get package installation location

### DISCUSSION THREAD

**Discussion by mtrajano** [mason-org/mason.nvim#258] [Q&A] [answered]
In order to set up the `node-debug2` adapter you need to pass the path to the [nodeDebug.js script](https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation#Javascript) which lives in the `node-debug2-adapter` package folder. I want to do this in a way that uses the configured installation root path. The way I'm doing this now is using `require('mason-core.path').package_prefix('node-debug2-adapter') .. '/out/src/nodeDebug.js'` but this feels like something internal to mason, not sure if I can rely on this. Another way would be to access the "currently set configuration", something like `require('mason.settings').current.install_root_dir .. '/packages/node-debug2-adapter/out/src/nodeDebug.js`, which also doesn't feel great since I'm accessing specific properties that are subject to change. That `current` property also feels internal to mason and the config structure is subject to change. Just wanted to get your thoughts on this, thanks!

**Comment by williamboman** [comment, marked as answer]
Hey! So the best way to gather the installation location of a package is via [`Package:get_install_path()`](https://github.com/williamboman/mason.nvim/blob/main/doc/reference.md#packageget_install_path). However, as you mention, you shouldn't really be reaching into packages directly as this is prone to errors. The `node-debug2-adapter` package will link an executable with the same name, for easy access. If you're using nvim-dap, try something like:

```lua
dap.adapters.node2 = {
  type = 'executable',
  command = 'node-debug2-adapter',
  args = {},
}
```

**Reply by mtrajano** [reply]
Thanks really handy to know thanks!
